modules/repofiles/update.go

```diff
@@ -576,26 +686,24 @@ func createCommitRepoActions(repo *models.Repository, gitRepo *git.Repository, o
 	actions := make([]*CommitRepoActionOptions, 0, len(optsList))
 
 	for _, opts := range optsList {
-		isNewRef := opts.OldCommitID == git.EmptySHA
-		isDelRef := opts.NewCommitID == git.EmptySHA
-		if isNewRef && isDelRef {
+		if opts.IsNewRef() && opts.IsDelRef() {
 			return nil, fmt.Errorf("Old and new revisions are both %s", git.EmptySHA)
 		}
 		var commits = &repo_module.PushCommits{}
-		if strings.HasPrefix(opts.RefFullName, git.TagPrefix) {
+		if opts.IsNewTag() {
```

Comment:
This introduced a bug. It should be `opts.IsTag()`.  #10663 will resolve the problem.